### DISCUSSION TITLE
Targeted Contribution

### DISCUSSION THREAD

**Discussion by ibahloul-ps** [PrestaShop/PrestaShop#37703] [Ideas]
Hello guys :wave:
For those who don't know me yet, I’m a Product Manager at PrestaShop SA, and I have the ownership of the open-source project.
Today, I’m excited to share some news with you!

We’re launching an experiment because we’re trying to find ways to optimize the validation process of contributions.
Our idea is to focus on specific contributions rather than trying to deal with every PR that affects every subject.
Our first contribution topic will be adding new hooks, as they look like a great candidate for several reasons:
They are simple both technically and functionally
They enhance PrestaShop’s extensibility by limiting overrides
They can be released in patch versions, offering near-instant value to merchants
We’ve updated [the documentation](https://devdocs.prestashop-project.org/8/development/components/hook/register-new-hook/) regarding hook contributions, and PrestaShop SA maintainers and committers will adjust the review process accordingly.
During this test, which is time-boxed (6 to 8 weeks), we will focus on PRs adding new hooks, and they will be released with the next minor version 8.2.X :rocket:

NB: We still have a limited number of people available to review & test, but we’re committed to improving ourselves with this new process.

Thank you for your contributions and support! Let’s make this a success together :muscle:

**Comment by Hlavtox** [comment]
Hey @ibahloul-ps, not really sure what this should achieve?

> Our idea is to focus on specific contributions rather than trying to deal with every PR that affects every subject.

So, only specific contributions will be validated in a given frame or what does this mean? And if somebody opens PR on something else, it must wait? I don't understand.

> During this test, which is time-boxed (6 to 8 weeks), we will focus on PRs adding new hooks, and they will be released with the next minor version 8.2.X 🚀

So, you want to add new hooks by force or what does that mean? If a hook (or any other thing) is needed, it's just added at whatever time. You think suddenly people will come up with dozens of hooks that are suddenly required?

This will only lead into people creating hooks that are not needed and not well thought out, like here - https://github.com/PrestaShop/PrestaShop/pull/37596.

**Reply by ibahloul-ps** [reply]
Hello @Hlavtox,
Thanks for your feedback!

> And if somebody opens PR on something else, it must wait? I don't understand.

Yes, during this test, we will prioritize the review of the PRs adding new hooks!

> You think suddenly people will come up with dozens of hooks that are suddenly required?

We don't know, let's hope so! That's the whole point of a test, to check whether an assumption is correct... or not!

**Reply by matks** [reply]
> So, you want to add new hooks by force or what does that mean? If a hook (or any other thing) is needed, it's just added at whatever time. You think suddenly people will come up with dozens of hooks that are suddenly required?

One of the aim of this idea is to get rid of overrides. We see here and there modules using overrides, and when we ask why, the answer is "There is a hook missing". We don't expect people to contribute random useless hooks but rather use this opportunity to submit the hooks they've been missing before, knowing we'll focus them.

**Reply by kpodemski** [reply]
In that case, it would be wise to communicate about this initiative to the authors from PS Marketplace whose solutions use overrides 👍🏻

**Reply by Hlavtox** [reply]
@ibahloul-ps Ilies, sorry, but you are burning time on problems that don't exist. I suggest on focusing on PrestaShop having a basic ecommerce system functionality. 

I love that some initiative is already running, for example multi-carrier orders by @MateoCastilloPS.

But, there is bambillion of BASIC things missing - implementing external invoicing hooks, external stock management hooks, implementing proper european VAT hooks, ability to change payment method of an order, change vat rates of orders, use conditions in emails, properly translate half of european languages, stock system magically creating pieces on returns and many other things reported over and over again. :(

**Reply by BenTen** [reply]
Of course there are a lot of problem to fix.

But I think, the way to override files instead than use hook is, for my opinion, a security issue and that increase trouble during migration, who help people to not upgrade shop.

The difficulty to upgrade without trouble is a big negative point of PS (in my opinion).

A lot of good developers don't help (or stopped to help) on github because they don't want work for a PR who will be checked 5 years after.

The fact is if some PR can be checked in priority, this kind of developer can be able to help more.
This comment is a good example for what I try to say: https://github.com/PrestaShop/PrestaShop/issues/38095#issuecomment-2672325937

I guess it doesn't mean all PR of hook will be accepted or integrated.
It doesn't mean other PR will be ignored.
It's just an idea to give opportunity to hook developers to improve the solution.

Why them and not others? Why not. Everyone can't be in first place.

**Comment by echar577gh** [comment]
Hi developers - Is it possible to add nginx and php services to the docker-compose.yml? That way we can ensure all four services - prestashop, mysql, php and nginx - will be in sync. Thank you for your consideration.

**Reply by Touxten** [reply]
Hello,

I think you need to create a new discussion
